File: ShellService/src/main/java/com/example/shellservce/Entities/SSHClient.java
package com.example.shellservce.Entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Table(name = "shhclient")
public class SSHClient {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "ssh_seq")
    @SequenceGenerator(name = "ssh_seq", sequenceName = "ssh_seq", allocationSize = 1)
    private Long id;

    private String name;

    private String username;

    private String host;
    @Column(name = "ssh_user")
    private String user;

    private String port;

    private String password;

    private String key;


}

File: web-shell-react/src/components/SSHClient.ts
export interface SSHClient{

    id:number|null,

    name:String|null,

    host:String|null;

    user:String|null;

    port:String|null;

    password:String|null;

    key:String | null;

    method: Method | null
}

export enum Method{
    password,key
}